src/components/description/DescriptionSubscriber.js

```diff
@@ -0,0 +1,38 @@
+import React, { useCallback, useEffect, useState } from 'react';
+import PubSub from 'pubsub-js';
+import { METADATA_ADD } from '../../events';
+import TitleInfo from '../TitleInfo';
+import Description from './Description';
+
+export default function DescriptionSubscriber(props) {
+  const { description, onReady, removeGridComponent } = props;
+
+  const onReadyCallback = useCallback(onReady, []);
+  const [metadata, setMetadata] = useState({});
+
+  const handleMetadataAdd = useCallback((msg, { name: layerName, metadata: layerMetadata }) => {
+    setMetadata({
+      ...metadata,
+      [layerName]: layerMetadata,
+    });
+  }, [metadata, setMetadata]);
+
+  useEffect(() => {
+    const metadataAddToken = PubSub.subscribe(METADATA_ADD, handleMetadataAdd);
+    onReadyCallback();
+    return () => PubSub.unsubscribe(metadataAddToken);
+  }, [onReadyCallback, handleMetadataAdd]);
```

Comment:
This might be personal preference, but in this context I would wrap the event handler within `useEffect` (sort of described [here](https://reactjs.org/docs/hooks-effect.html#tip-optimizing-performance-by-skipping-effects)). By using the callback in `setMetadata`, `metadata` doesn't need to be added to the dependency array of `useEffect`. I believe the eslint rules don't require `setMetadata` in the dependency array either since the setter is not stateful.

```suggestion
  useEffect(() => {
    function handleMetadataAdd(msg, { name: layerName, metadata: layerMetadata }) {
      setMetadata(prevMetadata => ({ ...prevMetadata, [layerName]: layerMetadata }));
    }
    const metadataAddToken = PubSub.subscribe(METADATA_ADD, handleMetadataAdd);
    onReadyCallback();
    return () => PubSub.unsubscribe(metadataAddToken);
  }, [onReadyCallback]);
```

I think this change would trigger the effect once, rather than subscribing and unsubscribing each time a `METADATA_ADD` event is fired (since`metadata` is a dependency of `handleMetadataAdd`, and `handleMetadataAdd` is a dependency of `useEffect` currently).

Reply:
Oh, this is much better! I did not realize you were supposed to pass a function to the setter if it depends on the previous state! Thanks!